manifests/init.pp

```diff
@@ -6,18 +6,25 @@
 # Parameters
 # ----------
 #
-# * `sample parameter`
-#   Explanation of what this parameter affects and what it defaults to.
-#   e.g. "Specify one or more upstream ntp servers as an array."
+# * `chroot_base: 
+#   Absolute path for chroot.By default, it uses /chroot. To set another, just write in hiera: sftp_jail::chroot_base: '/var/anotherchroot'
+#   Don't type the back stash
 #
 class sftp_jail (
   Stdlib::Absolutepath $chroot_base = '/chroot',
 ) inherits ::sftp_jail::params {
 
-  file { $chroot_base:
+# To make sure it creates recursively the directory
+  exec { "Create ${chroot_base}":
+    creates => $chroot_base,
+    command => "/bin/mkdir -p ${chroot_base}",
+  } -> file { $chroot_base: #}
+#  file { $chroot_base:
     ensure => 'directory',
     owner  => 'root',
     group  => 'root',
     mode   => '0755',
   }
+create_resources('sftp_jail::user', lookup('sftp_jail::user', {merge => 'deep'}))
```

Comment:
It's preferred create actual parameters and [set the merge behavior in hiera data](https://puppet.com/docs/puppet/5.4/hiera_merging.html#configure-merge-behavior-in-hiera-data).

Reply:
The parameter should probably be `users` (plural). I think iteration is preferred over `create_resources` too.

ie something a bit like...
```puppet
$users.each |String $key, Hash $attrs| {
  sftp_jail::user { $key:
    * => $attrs,
  }
}
```

Reply:
Removed create_resources in the latest commit.